sumologic/sumologic_cse_network_block.go

```diff
@@ -0,0 +1,83 @@
+package sumologic
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+)
+
+func (s *Client) GetCSENetworkBlock(id string) (*CSENetworkBlock, error) {
+	data, _, err := s.Get(fmt.Sprintf("sec/v1/network-blocks/%s", id), false)
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil {
+		return nil, nil
+	}
+
+	var response CSENetworkBlockResponse
+	err = json.Unmarshal(data, &response)
+	if err != nil {
+		return nil, err
+	}
+
+	return &response.CSENetworkBlock, nil
+}
+
+func (s *Client) DeleteCSENetworkBlock(id string) error {
+	_, err := s.Delete(fmt.Sprintf("sec/v1/network-blocks/%s", id))
+
+	return err
+}
+
+func (s *Client) CreateCSENetworkBlock(cseNetworkBlock CSENetworkBlock) (string, error) {
+
+	request := CSENetworkBlockRequest{
+		CSENetworkBlock: cseNetworkBlock,
+	}
+
+	var response CSENetworkBlockResponse
+
+	responseBody, err := s.Post("sec/v1/network-blocks", request, false)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal(responseBody, &response)
+
+	if err != nil {
+		return "", err
+	}
+
+	return response.CSENetworkBlock.ID, nil
+}
+
+func (s *Client) UpdateCSENetworkBlock(cseNetworkBlock CSENetworkBlock) error {
+	url := fmt.Sprintf("sec/v1/network-blocks/%s", cseNetworkBlock.ID)
+	log.Printf("[INFO] url: %s", url)
```

Comment:
remove this log line?

Reply:
done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/271/commits/4c00657834f23744c79da15a651cf03f35b3f5dc